### PR TITLE
Ansible role to detach and delete the disks from the node

### PR DESCRIPTION
**Pull request by nsathyaseelan**
Signed-off-by: sathyaseelan <sathyaseelan.n@cloudbyte.com>

**What this PR does / why we need it**:
  - This PR will be the ansible role to detach and delete the attached disk from each of the nodes in gke cluster.
  - And also modified the Disk attach role names.

@yudaykiran @ksatchit @dargasudarshan @gprasath can you review this PR.

### CODE REVIEW

**Comment by nsathyaseelan** [comment]
```
ansible-playbook 2.5.3
  config file = /home/sathyaseelan_n/e2e-sathya/openebs/e2e/ansible/ansible.cfg
  configured module search path = [u'/home/sathyaseelan_n/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.13 (default, Nov 24 2017, 17:33:09) [GCC 6.3.0 20170516]
Using /home/sathyaseelan_n/e2e-sathya/openebs/e2e/ansible/ansible.cfg as config file

PLAYBOOK: setup-disk-delete.yml ****************************************************************************************************************************************************************
1 plays in setup-disk-delete.yml

PLAY [localhost] *******************************************************************************************************************************************************************************

TASK [Gathering Facts] *************************************************************************************************************************************************************************
task path: /home/sathyaseelan_n/e2e-sathya/openebs/e2e/ansible/setup-disk-delete.yml:2
ok: [localhost]
META: ran handlers

TASK [k8s-gke-delete : Get the details of the available nodes] *********************************************************************************************************************************
task path: /home/sathyaseelan_n/e2e-sathya/openebs/e2e/ansible/roles/k8s-gke-delete/tasks/main.yml:2
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get nodes | grep gke | awk {'print $1'}", "delta": "0:00:01.627243", "end": "2018-08-01 19:05:21.734665", "rc": 0, "start": "2018-08-01 19:05:20.107422", "stderr": "", "stderr_lines": [], "stdout": "gke-sathya-cluster-default-pool-509f5e91-32k8\ngke-sathya-cluster-default-pool-509f5e91-d5hv\ngke-sathya-cluster-default-pool-509f5e91-mcd5", "stdout_lines": ["gke-sathya-cluster-default-pool-509f5e91-32k8", "gke-sathya-cluster-default-pool-509f5e91-d5hv", "gke-sathya-cluster-default-pool-509f5e91-mcd5"]}

TASK [k8s-gke-delete : include] ****************************************************************************************************************************************************************
task path: /home/sathyaseelan_n/e2e-sathya/openebs/e2e/ansible/roles/k8s-gke-delete/tasks/main.yml:9
included: /home/sathyaseelan_n/e2e-sathya/openebs/e2e/ansible/roles/k8s-gke-delete/tasks/include.yml for localhost
included: /home/sathyaseelan_n/e2e-sathya/openebs/e2e/ansible/roles/k8s-gke-delete/tasks/include.yml for localhost
included: /home/sathyaseelan_n/e2e-sathya/openebs/e2e/ansible/roles/k8s-gke-delete/tasks/include.yml for localhost

TASK [k8s-gke-delete : Getting the total number of disk in a nodes] ****************************************************************************************************************************
task path: /home/sathyaseelan_n/e2e-sathya/openebs/e2e/ansible/roles/k8s-gke-delete/tasks/include.yml:2
changed: [localhost -> None] => {"changed": true, "cmd": "gcloud compute disks list --filter=\"NAME:( gke-sathya-cluster-default-pool-509f5e91-32k8 )\" | grep disk | awk '{print $1}'", "delta": "0:00:05.298651", "end": "2018-08-01 19:05:27.374650", "rc": 0, "start": "2018-08-01 19:05:22.075999", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [k8s-gke-delete : Detaching the disk from the nodes] **************************************************************************************************************************************
task path: /home/sathyaseelan_n/e2e-sathya/openebs/e2e/ansible/roles/k8s-gke-delete/tasks/include.yml:9
skipping: [localhost] => {"changed": false, "results": [], "skipped_reason": "No items in the list"}

TASK [k8s-gke-delete : Getting the total number of disk in a nodes] ****************************************************************************************************************************
task path: /home/sathyaseelan_n/e2e-sathya/openebs/e2e/ansible/roles/k8s-gke-delete/tasks/include.yml:2
changed: [localhost -> None] => {"changed": true, "cmd": "gcloud compute disks list --filter=\"NAME:( gke-sathya-cluster-default-pool-509f5e91-d5hv )\" | grep disk | awk '{print $1}'", "delta": "0:00:04.547079", "end": "2018-08-01 19:05:32.276440", "rc": 0, "start": "2018-08-01 19:05:27.729361", "stderr": "", "stderr_lines": [], "stdout": "gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk0\ngke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk1", "stdout_lines": ["gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk0", "gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk1"]}

TASK [k8s-gke-delete : Detaching the disk from the nodes] **************************************************************************************************************************************
task path: /home/sathyaseelan_n/e2e-sathya/openebs/e2e/ansible/roles/k8s-gke-delete/tasks/include.yml:9
changed: [localhost -> None] => (item=gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk0) => {"changed": true, "cmd": "gcloud compute instances detach-disk gke-sathya-cluster-default-pool-509f5e91-d5hv --disk gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk0 --zone \"us-central1-a\" && gcloud compute disks delete \"gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk0\" --zone \"us-central1-a\"", "delta": "0:00:21.254772", "end": "2018-08-01 19:05:53.808817", "item": "gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk0", "rc": 0, "start": "2018-08-01 19:05:32.554045", "stderr": "Updated [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/instances/gke-sathya-cluster-default-pool-509f5e91-d5hv].\nThe following disks will be deleted:\n - [gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk0] in\n [us-central1-a]\n\nDo you want to continue (Y/n)?  \nDeleted [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/disks/gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk0].", "stderr_lines": ["Updated [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/instances/gke-sathya-cluster-default-pool-509f5e91-d5hv].", "The following disks will be deleted:", " - [gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk0] in", " [us-central1-a]", "", "Do you want to continue (Y/n)?  ", "Deleted [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/disks/gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk0]."], "stdout": "", "stdout_lines": []}
changed: [localhost -> None] => (item=gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk1) => {"changed": true, "cmd": "gcloud compute instances detach-disk gke-sathya-cluster-default-pool-509f5e91-d5hv --disk gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk1 --zone \"us-central1-a\" && gcloud compute disks delete \"gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk1\" --zone \"us-central1-a\"", "delta": "0:00:24.726788", "end": "2018-08-01 19:06:18.779124", "item": "gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk1", "rc": 0, "start": "2018-08-01 19:05:54.052336", "stderr": "Updated [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/instances/gke-sathya-cluster-default-pool-509f5e91-d5hv].\nThe following disks will be deleted:\n - [gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk1] in\n [us-central1-a]\n\nDo you want to continue (Y/n)?  \nDeleted [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/disks/gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk1].", "stderr_lines": ["Updated [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/instances/gke-sathya-cluster-default-pool-509f5e91-d5hv].", "The following disks will be deleted:", " - [gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk1] in", " [us-central1-a]", "", "Do you want to continue (Y/n)?  ", "Deleted [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/disks/gke-sathya-cluster-default-pool-509f5e91-d5hv-kdtscxbmhm-disk1]."], "stdout": "", "stdout_lines": []}

TASK [k8s-gke-delete : Getting the total number of disk in a nodes] ****************************************************************************************************************************
task path: /home/sathyaseelan_n/e2e-sathya/openebs/e2e/ansible/roles/k8s-gke-delete/tasks/include.yml:2
changed: [localhost -> None] => {"changed": true, "cmd": "gcloud compute disks list --filter=\"NAME:( gke-sathya-cluster-default-pool-509f5e91-mcd5 )\" | grep disk | awk '{print $1}'", "delta": "0:00:05.194975", "end": "2018-08-01 19:06:24.264619", "rc": 0, "start": "2018-08-01 19:06:19.069644", "stderr": "", "stderr_lines": [], "stdout": "gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk0\ngke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk1", "stdout_lines": ["gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk0", "gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk1"]}

TASK [k8s-gke-delete : Detaching the disk from the nodes] **************************************************************************************************************************************
task path: /home/sathyaseelan_n/e2e-sathya/openebs/e2e/ansible/roles/k8s-gke-delete/tasks/include.yml:9
changed: [localhost -> None] => (item=gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk0) => {"changed": true, "cmd": "gcloud compute instances detach-disk gke-sathya-cluster-default-pool-509f5e91-mcd5 --disk gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk0 --zone \"us-central1-a\" && gcloud compute disks delete \"gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk0\" --zone \"us-central1-a\"", "delta": "0:00:19.359362", "end": "2018-08-01 19:06:43.911157", "item": "gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk0", "rc": 0, "start": "2018-08-01 19:06:24.551795", "stderr": "Updated [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/instances/gke-sathya-cluster-default-pool-509f5e91-mcd5].\nThe following disks will be deleted:\n - [gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk0] in\n [us-central1-a]\n\nDo you want to continue (Y/n)?  \nDeleted [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/disks/gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk0].", "stderr_lines": ["Updated [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/instances/gke-sathya-cluster-default-pool-509f5e91-mcd5].", "The following disks will be deleted:", " - [gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk0] in", " [us-central1-a]", "", "Do you want to continue (Y/n)?  ", "Deleted [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/disks/gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk0]."], "stdout": "", "stdout_lines": []}
changed: [localhost -> None] => (item=gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk1) => {"changed": true, "cmd": "gcloud compute instances detach-disk gke-sathya-cluster-default-pool-509f5e91-mcd5 --disk gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk1 --zone \"us-central1-a\" && gcloud compute disks delete \"gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk1\" --zone \"us-central1-a\"", "delta": "0:00:22.600951", "end": "2018-08-01 19:07:06.737165", "item": "gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk1", "rc": 0, "start": "2018-08-01 19:06:44.136214", "stderr": "Updated [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/instances/gke-sathya-cluster-default-pool-509f5e91-mcd5].\nThe following disks will be deleted:\n - [gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk1] in\n [us-central1-a]\n\nDo you want to continue (Y/n)?  \nDeleted [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/disks/gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk1].", "stderr_lines": ["Updated [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/instances/gke-sathya-cluster-default-pool-509f5e91-mcd5].", "The following disks will be deleted:", " - [gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk1] in", " [us-central1-a]", "", "Do you want to continue (Y/n)?  ", "Deleted [https://www.googleapis.com/compute/v1/projects/strong-eon-153112/zones/us-central1-a/disks/gke-sathya-cluster-default-pool-509f5e91-mcd5-kdtscxbmhm-disk1]."], "stdout": "", "stdout_lines": []}
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************************************
localhost                  : ok=10   changed=6    unreachable=0    failed=0
```

File: e2e/ansible/roles/k8s-gke-detach/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+
+ansible_connection: local
+cluster_zone: "{{ lookup ('env','CLUSTER_ZONE') }}"
```
Comment (yudaykiran) [inline e2e/ansible/roles/k8s-gke-detach/defaults/main.yml]: CLUSTER_ZONE=ZONE

File: e2e/ansible/roles/k8s-gke-detach/tasks/detach-delete.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+---
+- name: Getting the total number of disk in a nodes
+  shell: gcloud compute disks list --filter="NAME:( {{ outer_item }} )" | grep disk | awk '{print $1}'
+  args:
+    executable: /bin/bash
+  delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+  register: disk
+
```
Comment (dargasudarshan) [inline e2e/ansible/roles/k8s-gke-detach/tasks/detach-delete.yml]: @nsathyaseelan How are we getting the list of disks attached? Are we getting disks list by finding the node name along with disk name?

Comment (nsathyaseelan) [reply e2e/ansible/roles/k8s-gke-detach/tasks/detach-delete.yml]: @dargasudarshan we are getting disk list by finding the node name.

**Pull request merged**